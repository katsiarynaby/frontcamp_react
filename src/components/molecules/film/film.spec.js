import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import Film from './index';

describe('Film', () => {
    it("Film should render correctly", () => {
        const wrapper = shallow(<Film
            onClickSearch={jest.fn()}
            getFilm={jest.fn()}
            match={{
                path: "/film/:id",
                url: "/film/424785",
                isExact: true,
                params: "{id: \"424785\"}"
            }}
            film={{
                id: 424785,
                title: "Transformers 7",
                tagline: "",
                vote_average: 0,
                vote_count: 0,
                release_date: "2019-06-26",
                poster_path: "https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg",
                overview: "Plot unknown.",
                budget: 0,
                revenue: 0,
                genres: [
                    "Science Fiction",
                    "Action",
                    "Adventure"
                ],
                runtime: null
            }}
        />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
