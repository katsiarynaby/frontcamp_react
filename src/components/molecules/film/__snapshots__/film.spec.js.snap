// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Film should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
